src/rpc_monitor.rs

```diff
@@ -0,0 +1,181 @@
+use std::future::Future;
+use std::time::Duration;
+
+use actix::prelude::{Actor, Addr, AsyncContext, Context, Handler, Message};
+use awc::Client;
+use serde::{Deserialize, Serialize};
+use tracing::{debug, info, warn};
+
+use crate::metrics::rpc_metrics as metrics;
+use crate::types::Commitment;
+
+#[derive(Serialize)]
+struct Param {
+    commitment: Commitment,
+}
+#[derive(Serialize)]
+struct Request<'a> {
+    jsonrpc: &'a str,
+    id: u64,
+    method: &'a str,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    params: Option<[Param; 1]>,
+}
+#[derive(Deserialize, Debug)]
+struct Wrap<T> {
+    #[serde(flatten)]
+    inner: Response<T>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "lowercase")]
+enum Response<T> {
+    Result(T),
+    Error(Box<serde_json::value::RawValue>),
+}
+
+#[derive(Message, Debug)]
+#[rtype(result = "()")]
+enum MonitorMessage {
+    SlotUpdated(u64),
+    HealthUpdated(bool),
+}
+
+pub struct RpcMonitor {
+    client: Client,
+    rpc_url: String,
+    id: u64,
+}
+
+impl RpcMonitor {
+    pub fn init(url: &str, client: Client) -> Addr<Self> {
+        let actor = Self::new(url, client);
+        actor.start()
+    }
+
+    fn new(url: &str, client: Client) -> Self {
+        RpcMonitor {
+            client,
+            rpc_url: url.to_owned(),
+            id: 1,
+        }
+    }
+
+    fn request_id(&mut self) -> u64 {
+        let next_id = self.id;
+        self.id += 1;
+        next_id
+    }
+
+    fn get_health(&mut self) -> impl Future<Output = anyhow::Result<bool>> {
+        let request = Request {
+            jsonrpc: "2.0",
+            id: self.request_id(),
+            method: "getHealth",
+            params: None,
+        };
+
+        let client = self.client.clone();
+        let rpc_url = self.rpc_url.clone();
+
+        async move {
+            Ok(
+                if Self::request::<_, String>(request, client, rpc_url).await? == "ok" {
+                    true
+                } else {
+                    false
+                },
+            )
+        }
```

Comment:
Только `use futures_util::future::TryFutureExt` нужен

```suggestion
        Self::request::<_, String>(request, client, rpc_url).map_ok(|resp| resp == "ok") 
```